Code/Framework/AzCore/Tests/Asset/AssetManagerLoadingTests.cpp

```diff
@@ -126,6 +133,202 @@ namespace UnitTest
+
+            SetupAssets();
+
+            AZStd::vector<AssetType> types;
+            m_assetHandlerAndCatalog->GetHandledAssetTypes(types);
+            for (const auto& type : types)
+            {
+                m_testAssetManager->RegisterHandler(m_assetHandlerAndCatalog, type);
+                m_testAssetManager->RegisterCatalog(m_assetHandlerAndCatalog, type);
+            }
+
+            {
+                AssetWithSerializedData asset6;
+
+                EXPECT_TRUE(m_streamerWrapper->WriteMemoryFile("TestAsset6.txt", &asset6, m_serializeContext));
+
+                AssetWithAssetReference asset1;
+                AssetWithAssetReference asset2;
+                AssetWithAssetReference asset3;
+                AssetWithAssetReference asset4;
+                AssetWithAssetReference asset5;
+
+                asset5.m_asset = m_testAssetManager->CreateAsset<AssetWithSerializedData>(MyAsset6Id, AssetLoadBehavior::QueueLoad);
+                EXPECT_TRUE(m_streamerWrapper->WriteMemoryFile("TestAsset5.txt", &asset5, m_serializeContext));
+
+                asset4.m_asset = m_testAssetManager->CreateAsset<AssetWithAssetReference>(MyAsset5Id, AssetLoadBehavior::QueueLoad);
+                EXPECT_TRUE(m_streamerWrapper->WriteMemoryFile("TestAsset4.txt", &asset4, m_serializeContext));
+
+                asset3.m_asset = m_testAssetManager->CreateAsset<AssetWithAssetReference>(MyAsset4Id, AssetLoadBehavior::QueueLoad);
+                EXPECT_TRUE(m_streamerWrapper->WriteMemoryFile("TestAsset3.txt", &asset3, m_serializeContext));
+
+                asset2.m_asset = m_testAssetManager->CreateAsset<AssetWithAssetReference>(MyAsset3Id, AssetLoadBehavior::QueueLoad);
+                EXPECT_TRUE(m_streamerWrapper->WriteMemoryFile("TestAsset2.txt", &asset2, m_serializeContext));
+
+                asset1.m_asset = m_testAssetManager->CreateAsset<AssetWithAssetReference>(MyAsset2Id, AssetLoadBehavior::QueueLoad);
+                EXPECT_TRUE(m_streamerWrapper->WriteMemoryFile("TestAsset1.txt", &asset1, m_serializeContext));
+
+                m_assetHandlerAndCatalog->m_numCreations = 0;
+            }
+        }
+
+        void CheckFinishedCreationsAndDestructions()
+        {
+            // Make sure asset jobs have finished before validating the number of destroyed assets, because it's possible that the asset job
+            // still contains a reference on the job thread that won't trigger the asset destruction until the asset job is destroyed.
+            BlockUntilAssetJobsAreComplete();
+            m_testAssetManager->DispatchEvents();
+            EXPECT_EQ(m_assetHandlerAndCatalog->m_numCreations, m_assetHandlerAndCatalog->m_numDestructions);
+        }
+    };
+
+    TEST_F(AssetReloading, AssetReloadTest)
+    {
+        // Test OnAssetDependencyReloaded notification is sent all the way to the parent asset up a deep chain of dependencies
+        const auto timeoutSeconds = AZStd::chrono::seconds(200);
```

Comment:
Yeah, I agree with that.
This test shouldn't ever get anywhere close to 200 seconds.

Reply:
Oops, I bumped that up since it was making debugging very difficult and forgot to reset it.  It should be 10 seconds like the rest of the tests, I'll reset it.  Good catch.